autoscaler/controllers/nodecollector/collectorconfig/spanmetrics.go

```diff
@@ -0,0 +1,159 @@
+	spanMetricsTracesInConnectorName                 = "forward/trace/spanmetrics"
+	spanMetricsPipelineName                          = "traces/spanmetrics"
+	spanMetricsResourceRemoveDimensionsProcessorName = "resource/spanmetrics/remove-dimensions"
+)
+
+func getSpanMetricsConnectorConfig(spanMetricsConfig common.MetricsSourceSpanMetricsConfiguration) config.GenericMap {
+	histogramConfig := config.GenericMap{}
+
+	if spanMetricsConfig.HistogramDisabled {
+		histogramConfig["disabled"] = true
+	} else {
+		if spanMetricsConfig.ExplicitHistogramBuckets != nil {
+			histogramConfig["explicit"] = config.GenericMap{
+				"buckets": spanMetricsConfig.ExplicitHistogramBuckets,
+			}
+		}
+	}
+
+	dimensionsAttributeNames := []string{
+		"http.method",
+		"http.request.method",
+		"http.status_code",
+		"http.response.status_code",
+		"http.route",
+	}
+	if spanMetricsConfig.AdditionalDimensions != nil {
+		dimensionsAttributeNames = append(dimensionsAttributeNames, spanMetricsConfig.AdditionalDimensions...)
+	}
+
+	dimensions := make([]config.GenericMap, len(dimensionsAttributeNames))
+	for i, dimensionAttributeName := range dimensionsAttributeNames {
+		dimensions[i] = config.GenericMap{
+			"name": dimensionAttributeName,
+		}
+	}
+
+	return config.GenericMap{
+		"histogram": histogramConfig,
+		// Taking into account changes in the semantic conventions, to support a range of instrumentation libraries
+		"dimensions": dimensions,
+		"exemplars": config.GenericMap{
+			"enabled": true,
+		},
+		"exclude_dimensions":              []string{"status.code"},
+		"dimensions_cache_size":           1000,
+		"aggregation_temporality":         "AGGREGATION_TEMPORALITY_CUMULATIVE",
+		"metrics_flush_interval":          spanMetricsConfig.Interval,
+		"metrics_expiration":              spanMetricsConfig.MetricsExpiration,
+		"resource_metrics_key_attributes": []string{"service.name", "telemetry.sdk.language", "telemetry.sdk.name"},
+		"events": config.GenericMap{
+			"enabled": true,
+			"dimensions": []config.GenericMap{
+				{
+					"name": "exception.type",
+				},
+				{
+					"name": "exception.message",
+				},
+			},
+		},
+	}
+}
+
+func getSpanMetricsConnectors(spanMetricsConfig common.MetricsSourceSpanMetricsConfiguration) config.GenericMap {
+	return config.GenericMap{
+		spanMetricsConnectorName:         getSpanMetricsConnectorConfig(spanMetricsConfig),
+		spanMetricsTracesInConnectorName: &config.GenericMap{},
+	}
+}
+
+func getSpanMetricsPipelineProcessors(spanMetricsConfig common.MetricsSourceSpanMetricsConfiguration) (config.GenericMap, []string) {
+
+	processors := config.GenericMap{}
+	processorNames := []string{}
+	resourceAttrToExclude := []string{
+		// always delete these two attributes, as they are just noise in span metrics
+		// TODO: consider making it an opt-in configuration option one day
+		string(semconv.TelemetrySDKNameKey),
+		string(semconv.TelemetrySDKVersionKey),
+	}
```

Comment:
I'm not sure about excluding these by default, I think sdk.name is kind of nominally important in otel, though I don't know exactly how that relates to span metrics. if these are causing issues I'm fine with it but just thinking users can exclude them anyway if it's just annoying